src/connectivity/mod.rs

```diff
@@ -14,6 +15,9 @@ pub trait Connectivity: fmt::Debug {
 
     /// Disconnects from a wireless network currently connected to.
     fn disconnect(&self) -> Result<bool, WifiConnectionError>;
+
+    // determines speed when connected to a network
+    fn speed(&self) -> Result<String, WifiConnectionError>;
```

Comment:
Provide a default implementation of `unimplemented!()`. This should prevent the build from failing for other targets.